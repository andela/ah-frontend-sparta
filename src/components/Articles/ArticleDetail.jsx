import React from 'react';
import PropTypes, { shape } from 'prop-types';
import moment from 'moment';
import { Link } from 'react-router-dom';
import Comments from '../../pages/Comments';
import DeleteArticleComponentModel from './DeleteArticleModel';
import avartaImage from '../../assets/images/avarta.png';
import './article.scss';

const ArticleDetail = ({
  article,
  deleteAnArticle,
  comments,
  showReplies,
  replyDisplayState,
  postReplyToComment,
  onCommentChanged,
  postComment,
  deleteComment,
  updateComment,
  onChangeComment,
}) => (
  <div className="container article-detail-container">
    <div className="row">
      <div className="col-md-12">
        <div className="card">
          <div className="card-body customise-card-body">
            <h1 className="card-title article-title ">{article.title}</h1>
            <div className="row">
              <div className="col-md-3">
                <img src={avartaImage} className="img-circle" alt="user-icon" width="70px" height="70px" />
              </div>
            </div>
            <br />
            <button type="button" className="btn btn-sm btn-outline-secondary">Unfollow</button>
            &nbsp;
            &nbsp;
            <button type="button" className="btn btn-sm btn-outline-secondary">Follow</button>
            <p>
              {article.author && (
                <b>
                  Created by
                  {' '}
                  { article.author.username}
                </b>
              )}
            </p>
            <h6 className="card-subtitle mb-2 text-muted">
              {
                              article.createdAt && moment(article.createdAt.slice(0, 10)).format(
                                'LL',
                              )
                  }

            </h6>
            <br />
            <p className="card-text article-text">
              { article.body }
            </p>
            <br />
            <br />
            <br />
            <p className="show-tags">
                Tags
                &nbsp;
                &nbsp;
              {article.tags && article.tags.map(tag => (
                <span
                  className="badge badge-light badge-color"
                >
                  {tag}
                </span>
              ))}
            </p>
            {article.author && article.author.username === localStorage.getItem('username')
              ? (
                <div className="article-btns">
                  <Link
                    to={{
                      pathname: `/articles/edit/${article.slug}`,
                    }}
                    className="btn btn-primary float-left"
                  >
                    Edit
                    {' '}
                    <i className="far fa-edit" />
                  </Link>
                  <button
                    type="button"
                    className="btn btn-danger float-right"
                    data-toggle="modal"
                    data-target="#deleteArticleModal"
                  >
                Delete
                    {' '}
                    <i className="fas fa-trash-alt" />
                  </button>
                </div>
              )
              : ''

            }
            <DeleteArticleComponentModel
              title={article.title}
              slug={article.slug}
              username={
              article.author && article.author.username}
              deleteAnArticle={deleteAnArticle}
            />
            <Comments comments={comments} showReplies={showReplies} replyDisplayState={replyDisplayState} slug={article.slug} postReplyToComment={postReplyToComment} onCommentChanged={onCommentChanged} postComment={postComment} deleteComment={deleteComment} updateComment={updateComment} onChangeComment={onChangeComment} />
          </div>
        </div>
      </div>
    </div>
  </div>
);

ArticleDetail.defaultProps = {
  article: shape(PropTypes.object.isRequired),
};
ArticleDetail.propTypes = {
  article: shape(PropTypes.object.isRequired),
};

export default ArticleDetail;
