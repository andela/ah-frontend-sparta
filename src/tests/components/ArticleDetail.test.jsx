import React from 'react';
import { shallow } from 'enzyme';
import ArticleDetail from '../../components/Articles/ArticleDetail';

describe('Articles Details Component', () => {
  let mountApp;
  const props = {
    article: {
      id: 1,
      author: {
        username: 'rogha',
      },
      article_read_time: 'less than a minute read',
      title: 'Post one',
      description: 'too long a story',
      slug: 'post-one-first',
      body: 'ahhhhh --edit me',
      createdAt: '2019-04-18T11:55:58.111843Z',
      tags: ['Arsenal', 'soccer', 'Andela'],
    },
  };
  beforeEach(() => {
    mountApp = shallow(<ArticleDetail {...props} />);
  });

  it('Should render without crashing', () => {
    expect(mountApp).toMatchSnapshot();
  });

  it('Should render articles with tags', () => {
    expect(mountApp.find('span')).toHaveLength(3);
  });
});
