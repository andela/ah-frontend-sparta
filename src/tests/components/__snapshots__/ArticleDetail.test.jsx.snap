// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Details Component Should render without crashing 1`] = `ShallowWrapper {}`;
